src/Aspire.ProjectTemplates/templates/aspire-apphost-singlefile/.template.config/template.json

```diff
@@ -0,0 +1,83 @@
+{
+  "$schema": "https://json.schemastore.org/template",
+  "author": "Microsoft",
+  "classifications": [
+    "Common",
+    "Aspire",
+    "AppHost",
+    "SingleFile"
+  ],
+  "name": "Aspire Single-File App Host",
+  "shortName": "aspire-apphost-singlefile",
+  "sourceName": "SingleFileAppHost",
+  "defaultName": "AspireSingleFileHost",
+  "description": "A minimal Aspire AppHost for single-file scenarios (Aspire 9.5+, .NET 10+).",
+  "identity": "Aspire.AppHost.SingleFile.CSharp.10.0",
+  "groupIdentity": "Aspire.AppHost.SingleFile",
+  "precedence": "8900",
+  "tags": {
+    "language": "C#",
+    "type": "project"
+  },
+  "sources": [
```

Comment:
@mitchdenny I think we should include a `"constraints"` property here that forcibly hides this template from VS and `dotnet new`.  See https://github.com/dotnet/templating/wiki/Reference-for-template.json#template-constraints and https://github.com/dotnet/templating/wiki/Constraints#template-engine-host for details

Reply:
Alternatively we might consider putting this in a separate template package and teaching the Aspire CLI to install it to a custom hive so that it doesn't show up in VS or the `dotnet` CLI.

Reply:
I do think it would be useful to be able to have it in `dotnet` CLI as a diagnostic mechanism.

Reply:
Then we'd want to at least limit it to the .NET 10 SDK via a constraint.